activejob/lib/active_job/callbacks.rb

```diff
@@ -27,16 +28,33 @@ class << self
     end
 
     included do
-      define_callbacks :perform
-      define_callbacks :enqueue
+      class_attribute :return_false_on_aborted_enqueue, instance_accessor: false, instance_predicate: false, default: false
```

Comment:
Those are class attributes but I'm not seeing a test to make sure that if we change the value of those configs in a subclass it will not affect all the other classes. Can you write a test like that?

Reply:
Overriding `return_false_on_aborted_enqueue` in a subclass won't have the effect you think. It's not possible to have a job run after_enqueue callbacks while others don't. It's all jobs or no jobs.

When AJ define callbacks, they are stored in a class attribute https://github.com/rails/rails/blob/737fb59eb358b1cb68001257ec9274d0c9f8977a/activesupport/lib/active_support/callbacks.rb#L69

but this class_attribute is not meant to be override. All subclasses of ActiveJob::Base shares the same callbacks and thus we can't modify a callback for a particular job.

Reply:
> Maybe it's better to make it a `cattr_accessor` instead ?

Yes. If you can't have per job config it is better to use `cattr_accessor`.